docs/site/src/content/docs/contributing/new-contributors.mdx

```diff
@@ -0,0 +1,192 @@
+   
+   ```bash
+   $ git remote -v
+   origin  https://github.com/YOUR_USERNAME/player.git (fetch)
+   origin  https://github.com/YOUR_USERNAME/player.git (push)
+   upstream        https://github.com/player-ui/player.git (fetch)
+   upstream        https://github.com/player-ui/player.git (push)
+   ```
+
+3. **Set Up Development Environment**
+   
+   Follow the [requirements setup](/contributing/#requirements) and install the necessary tools and dependencies.
+
+4. **Choose Your Platform**
+   
+   <CardGrid>
+     <Card title="Core & Web Development" icon="seti:javascript">
+       TypeScript/JavaScript plugin development
+     </Card>
+     <Card title="Android Development" icon="seti:kotlin">
+       Native Android development
+     </Card>
+     <Card title="iOS Development" icon="seti:swift">
+       Native iOS development
+     </Card>
+   </CardGrid>
+
+5. **Need Help?**
+   
+   Check the [Troubleshooting Guide](/contributing/troubleshooting) for common issues and solutions.
+
+</Steps>
+
+## Keeping Your Fork Updated
+
+To keep your fork in sync with the main repository, you can leverage the upstream remote that was created when you set up your fork:
+
+```bash
+# Make sure we're on main branch first
+git checkout main
+git pull upstream main
+git push origin main
+```
+
+This allows you to quickly take everything that the main repo has on main and push it back up to your fork. You should do this anytime you want to contribute to ensure you don't have a stale branch with outdated files.
+
+> **Alternative Method**: You can also create forks and PRs through [GitHub's web interface](https://github.com/player-ui/player) if you prefer not to use the CLI.
+
+
+## Making Your First Contribution
+
+### Good First Issues
+
+Look for issues labeled with:
+- <Badge text="good first issue" variant="tip" /> - Perfect for newcomers
+- <Badge text="help wanted" variant="note" /> - Community contributions welcome
+- <Badge text="documentation" variant="caution" /> - Great for getting familiar with the codebase
+
+### Development Workflow
+
+<Steps>
+
+1. **Create a Branch**
+
+   ```bash
+   git checkout -b feature/your-feature-name
+   ```
+
+2. **Make Your Changes**
+
+   - Write your code
+   - Add or update tests
+   - Update documentation if needed
+
+3. **Test Your Changes**
+
+   ```bash
+   # Run relevant tests
+   bazel test //path/to/your:tests
+
+   # For broader testing
+   bazel test //...
+   ```
+
+4. **Commit Your Changes**
+
+   ```bash
+   git add .
+   git commit -S -m "feat: add your feature description"
+   ```
+
+   > Note: The `-S` flag signs your commit, which is required for this project.
+
+5. **Push and Create Pull Request**
+
+   ```bash
+   git push origin feature/your-feature-name
+   ```
+
+   When you're ready to submit a Pull Request, there's a great CLI command for that too:
+
+   ```bash
+   gh pr create
```

Comment:
Unless we're going to recommend people download the github cli we probably don't want to have those specific commands here